### DISCUSSION TITLE
`to_agraph` function modifies attributes of source graph

### DISCUSSION THREAD

**Discussion by ashwin-nayak** [networkx/networkx#7608] [Q&A] [answered]
Consider a simple graph, 

```python
import networkx as nx
G = nx.Graph()
G.add_node(1, pos=(0, 0))
G.add_node(2, pos=(1, 0))
G.add_edge(1, 2)
```
And say, one wants to convert to PyGraphViz AGraph Class using `to_agraph` interface,

```python
H = nx.nx_agraph.to_agraph(G)
```

This routine formats the attribute `pos` for each node to conform to the GraphViz type. However, the formatting is done *in-place*. 

```python
print(H.get_node(1).attr['pos'], G.nodes[1]['pos'])
# prints ('0,0!', '0,0!')
```

For now, I address this by `H = nx.nx_agraph.to_agraph(G.copy())`, but is this behavior intended?

**Comment by dschult** [comment]
Thank you for reporting this Bug!

It looks like PR #6474 changes the NetworkX graph's attributes to be in the dot format.  That is suprising!

I would expect it to reformat the pygraphviz.AGraph attributes after AGraph is created -- or, even better, reformat the pos values when first written to the AGraph. There is no reason to change the original graph that I can see.  We'll work to fix that.

As a workaround until we get the fix in, you can make a copy of the graph before using `to_agraph`. 

If you'd like to make an attempt to correct the error, it appears at https://github.com/networkx/networkx/blob/ae63d89bf90b6244fec5e6590d36990659c99fda/networkx/drawing/nx_agraph.py#L141
where we convert the G node attribute "pos" format. It seems like that reformatting should occur around like 160 where the nodes are added.

**Reply by ashwin-nayak** [reply]
Thank you. You are right - I will then open an issue and submit a pull request fixing this.

**Comment by mjschwenne** [comment, marked as answer]
Fixed in #7610
